src/Components/Aspire.Orleans.Server/AspireOrleansServerExtensions.cs

```diff
@@ -0,0 +1,214 @@
+
+            if (serverSettings.Reminders is { } remindersSection && remindersSection.Exists())
+            {
+                ApplyRemindersSettings(builder, siloBuilder, remindersSection);
+            }
+
+            // Enable distributed tracing for open telemetry.
+            siloBuilder.AddActivityPropagation();
+        });
+
+        return builder;
+    }
+
+    private static void ApplyGrainStorageSettings(IHostApplicationBuilder builder, ISiloBuilder siloBuilder, string name, IConfigurationSection configuration)
+    {
+        var connectionSettings = new ConnectionSettings();
+        configuration.Bind(connectionSettings);
+
+        var type = connectionSettings.ConnectionType;
+        var connectionName = connectionSettings.ConnectionName;
+
+        if (string.IsNullOrWhiteSpace(type))
+        {
+            throw new ArgumentException(message: $"A \"ConnectionType\" value must be specified for \"GrainStorage\" named '{name}'.", innerException: null);
+        }
+
+        if (string.IsNullOrWhiteSpace(connectionName))
+        {
+            throw new ArgumentException(message: $"A \"ConnectionName\" value must be specified for \"GrainStorage\" named '{name}'.", innerException: null);
+        }
+
+        if (string.Equals(InternalType, type, StringComparison.OrdinalIgnoreCase))
+        {
+            siloBuilder.AddMemoryGrainStorage(name);
+        }
+        else if (string.Equals(AzureTablesType, type, StringComparison.OrdinalIgnoreCase))
+        {
+            // Configure a table service client in the dependency injection container.
+            builder.AddKeyedAzureTableService(connectionName);
+
+            // Configure Orleans to use the configured table service client.
+            siloBuilder.AddAzureTableGrainStorage(name, optionsBuilder => optionsBuilder.Configure(
+                (AzureTableStorageOptions options, IServiceProvider serviceProvider) =>
+                {
+                    var tableServiceClient = Task.FromResult(serviceProvider.GetRequiredKeyedService<TableServiceClient>(connectionName));
+                    options.ConfigureTableServiceClient(() => tableServiceClient);
+                }));
+        }
+        else if (string.Equals(AzureBlobsType, type, StringComparison.OrdinalIgnoreCase))
+        {
+            // Configure a blob service client in the dependency injection container.
+            builder.AddKeyedAzureBlobService(connectionName);
+
+            // Configure Orleans to use the configured table service client.
+            siloBuilder.AddAzureBlobGrainStorage(name, optionsBuilder => optionsBuilder.Configure(
+                (AzureBlobStorageOptions options, IServiceProvider serviceProvider) =>
+                {
+                    var tableServiceClient = Task.FromResult(serviceProvider.GetRequiredKeyedService<BlobServiceClient>(connectionName));
+                    options.ConfigureBlobServiceClient(() => tableServiceClient);
+                }));
+        }
```

Comment:
This can't imply azure, so azure needs to be in the name if this requires azure specific clients.